source/simulator/introspection.cc

```diff
@@ -487,6 +520,22 @@ namespace aspect
   }
 
 
+
+  template <int dim>
+  bool
+  Introspection<dim>::is_composition_component (const unsigned int component_index) const
+  {
+    // All compositions live at the end. Just to be sure, there are no other components
+    // in our system after compositional fields, right?
+    Assert(component_indices.compositional_fields[0] > component_indices.temperature
+           && (*component_indices.compositional_fields.end()) == n_components-1, ExcInternalError());
```

Comment:
Wait what? You're dereferencing the `end()` iterator? I must be missing something, or this is not likely going to go well...

Reply:
Oops.

Reply:
OK, so I wasn't totally crazy? :-)